scripts/abi_artifact_generator/templates/setup_arguments.S.jinja_template

```diff
@@ -0,0 +1,53 @@
+/*#-
+This template file is distributed under the MIT License. See LICENSE.md for details.
+#*/
+
+/*(- generation_notice )*/
+
+/*{ if current_style == 'gcc' -}*/
+#include "../abi.h"
+
+#if defined(TARGET_x86_64) || defined(TARGET_i386)
+.intel_syntax noprefix
+#endif
+/*{- elif current_style == 'msvc' -}*/
+ifndef rax
+.model flat, c
```

Comment:
It declares calling convention, `flat, c` means `cdecl`.

Which is one of the big problem we'd have to solve if we plan to go full PE/COFF binary. This was easily avoidable in the Frankenstein approach.

Either way, out of scope for the current PR. Will do the comment though.

Reply:
> one of the big problem we'd have to solve if we plan to go full PE/COFF binary

Please explain

Reply:
For some godforsaken reason, microsoft-flavoured assembly encodes ABI data in. So, a `cdecl` function (`_name`) is not compatible with a `stdcall` function (`_name@4`) which, in turn, is not compatible with a `fastcall` function (`@name@4`).

This "model" is what encodes it for the current file.

In the "Frankenstein" approach, we don't care much, since this doesn't impact instructions - and we only "copy and paste" those into gcc context. If we want a full PE/COFF binary, we'd have to extend templates to actually support all of these fully. Should be doable, but is probably a bit of a PITA.

Reply:
Gotcha.

Reply:
Done.